src/xcode/ENA/ENA/Source/Client/Client.swift

```diff
@@ -187,9 +230,37 @@ extension Client {
 		}
 	}
 
+	func fetchDays(
+		_ days: [String],
+		hours: [Int],
+		of day: String,
+		country: String,
+		completion completeWith: @escaping DaysAndHoursCompletionHandler
+	) {
+		let group = DispatchGroup()
+		var hoursResult = HoursResult(errors: [], bucketsByHour: [:], day: day)
+		var daysResult = DaysResult(errors: [], bucketsByDay: [:])
+
+		group.enter()
+		fetchDays(days, forCountry: country) { result in
+			daysResult = result
+			group.leave()
+		}
+
+		group.enter()
+		fetchHours(hours, day: day, country: country) { result in
+			hoursResult = result
+			group.leave()
+		}
+		group.notify(queue: .main) {
+			completeWith(FetchedDaysAndHours(hours: hoursResult, days: daysResult))
+		}
```

Comment:
will fetchDays/fetchHours time out at some point or can we get stuck here?

Reply:
It will never get stuck. The dispatch group has no timeout and could get stuck but the URL session has one and the requests will time out and then the group will also "finish".